src/Symfony/Bundle/FrameworkBundle/Resources/config/validator.php

```diff
@@ -109,5 +159,259 @@
+            ])
+
+        ->set('validator.date_time', DateTimeValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => DateTimeValidator::class,
+            ])
+
+        ->set('validator.divisible_by', DivisibleByValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => DivisibleByValidator::class,
+            ])
+
+        ->set('validator.equal_to', EqualToValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => EqualToValidator::class,
+            ])
+
+        ->set('validator.expression_language_syntax', ExpressionLanguageSyntaxValidator::class)
+            ->args([service('validator.expression_language')->nullOnInvalid()])
+            ->tag('validator.constraint_validator', [
+                'alias' => ExpressionLanguageSyntaxValidator::class,
+            ])
+
+        ->set('validator.expression_syntax', ExpressionSyntaxValidator::class)
+            ->args([service('validator.expression_language')->nullOnInvalid()])
+            ->tag('validator.constraint_validator', [
+                'alias' => ExpressionSyntaxValidator::class,
+            ])
+
+        ->set('validator.file', FileValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => FileValidator::class,
+            ])
+
+        ->set('validator.greater_than', GreaterThanValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => GreaterThanValidator::class,
+            ])
+
+        ->set('validator.greater_than_or_equal', GreaterThanOrEqualValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => GreaterThanOrEqualValidator::class,
+            ])
+
+        ->set('validator.hostname', HostnameValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => HostnameValidator::class,
+            ])
+
+        ->set('validator.iban', IbanValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => IbanValidator::class,
+            ])
+
+        ->set('validator.identical_to', IdenticalToValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => IdenticalToValidator::class,
+            ])
+
+        ->set('validator.image', ImageValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => ImageValidator::class,
+            ])
+
+        ->set('validator.ip', IpValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => IpValidator::class,
+            ])
+
+        ->set('validator.isbn', IsbnValidator::class)
+            ->tag('validator.constraint_validator', [
+                'alias' => IsbnValidator::class,
+            ])
+
+        ->set('validator.is_false', IsFalseValidator::class)
```

Comment:
Are we sure to want to maintain this big list of services? Maybe the fix should be on the translator component since this is the only issue we founded due to not declared constraints?

Reply:
I agree with that, this PR is a glimpse of the involvements of a solution going in the way of the original feature taking the postulate that validators were all services

Reply:
I’m not sure since the issue is proof that `validator.constraint_validator` is unreliable. If a fix is applied to the translator component I think this tag would have to be removed.

Reply:
Alternative solution in the translator component available [here](https://github.com/symfony/symfony/pull/49785)

Reply:
Yeah #49785 feels better indeed; it’s not that `validator.constraint_validator` is unreliable, it’s just that unregistered validators cannot be tagged. But would it be possible to still consider it in `TranslatorPass` to take userland validators into account?

Reply:
Answering the question [here](https://github.com/symfony/symfony/pull/49785#issuecomment-1482879533) and closing this PR.